cli/src/runner.rs

```diff
@@ -0,0 +1,179 @@
+use std::io;
+use std::ops::Deref;
+use std::path::PathBuf;
+
+use analyzer::diagnostic::Diagnostic;
+use analyzer::engine::Engine;
+use analyzer::importer::ASTImporter;
+use analyzer::name::Name;
+use analyzer::relations::{Definition, SourceId};
+use analyzer::steps::typing::apply_types;
+use analyzer::types::engine::CodeEntry;
+use ast::group::Block;
+use ast::Expr;
+use compiler::compile;
+use context::source::{Source, SourceSegmentHolder};
+use lexer::lexer::lex;
+use parser::err::ParseReport;
+use parser::parse;
+
+use crate::cli::{display_exprs, display_tokens, execute, Configuration};
+use crate::display_bytecode::display_bytecode;
+use crate::formatted_diagnostic::render_diagnostic;
+use crate::formatted_parse_error::render_parse_error;
+use crate::source_importer::FileSourceImporter;
+
+pub(crate) fn run(source: PathBuf, working_dir: PathBuf, config: Configuration) -> bool {
```

Comment:
The returned bool is inverted, did you mean to return true in case of success? If it is for the `exit(code as i32)`, it should be inverted here.